### ISSUE TITLE
Fix `selector-type-no-unknown` false positives for `<custom-ident>` passed to pseudo-elements

### ISSUE THREAD

**Issue by jcayzac** [stylelint/stylelint#7911]
### What minimal example or steps are needed to reproduce the bug?

```css
html::view-transition-group(lightdark) {
	animation-timing-function:step-end;
}

:root::view-transition-old(lightdark) {
	animation-name: keep-old;
	z-index: 1;
}

:root::view-transition-new(lightdark) {
	animation-name: keep-new;
	z-index: 9999;
}
```

### What minimal configuration is needed to reproduce the bug?

```js
export default {
	extends: [
		'stylelint-config-standard',
		'stylelint-config-html',
	],
	rules: {
	},
}
```

### How did you run Stylelint?

from astro

### Which Stylelint-related dependencies are you using?

```
"stylelint": "^16.8.1",
"stylelint-config-html": "^1.1.0",
"stylelint-config-standard": "^36.0.1",
```

### What did you expect to happen?

No error, since this is [valid CSS syntax](https://developer.mozilla.org/en-US/docs/Web/CSS/::view-transition-group#custom-ident).

### What actually happened?

`Unexpected unknown type selector "lightdark" (selector-type-no-unknown)`

### Do you have a proposal to fix the bug?

Accept any identifier in those pseudo selectors, since they can be used by the Javascript-based View Transitions API.

**Comment by Mouvedia** [comment]
We already have code to ignore certain pseudo-classes arguments.
e.g. `odd`

https://github.com/stylelint/stylelint/blob/9897bf1f712a4788fdffad63a023a93ad67845e1/lib/utils/isStandardSyntaxTypeSelector.mjs#L15-L21

We should have the same for these pseudo-elements.

---

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by Mouvedia** [comment]
`view-transition-name` should be added to [this set](https://github.com/stylelint/stylelint/blob/main/lib/rules/custom-property-no-missing-var-function/index.mjs#L18) at the same time.
ref #5742

**Comment by Mouvedia** [comment]
[`::highlight()`](https://developer.mozilla.org/en-US/docs/Web/CSS/::highlight) is another pseudo-element which produces false positives.
ref w3c/csswg-drafts#9483

**Comment by romainmenke** [comment]
Side note: 

I've opened [a separate issue](https://github.com/stylelint/stylelint/issues/7915) to discuss a more general purpose helper so that we can have more consistent support across multiple rules for such functional pseudo's.

**Comment by Mouvedia** [comment]
`::part()`should be ignored too.
